staging/src/k8s.io/endpointslice/topologyheuristics/proportionalzonecpuheuristic.go

```diff
@@ -34,8 +34,11 @@ const (
 	overloadThreshold float64 = 0.2
 )
 
-// TopologyCache tracks the distribution of Nodes and endpoints across zones.
-type TopologyCache struct {
+// ProportionalZoneCPUHeuristic is a heuristic that provides EndpointSlice
+// topology hints using the [Proportional CPU Heuristic].
+//
+// [Proportional CPU Heuristic]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2433-topology-aware-hints#proportional-cpu-heuristic
+type ProportionalZoneCPUHeuristic struct {
```

Comment:
you should add `var _ Somether = (*MyType)(nil)` so the compiler validates it satistfy the interface